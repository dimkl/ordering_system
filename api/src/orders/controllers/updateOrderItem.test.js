/**
 * @integration-test true
 * @data-factory true
 */
const uuid = require('uuid');
const Order = require("../models/order");

describe("PATCH /order_items/:order_item_id", () => {
  beforeAll(() => require('../../shared/setupModels')());
  beforeEach(() => Order.knex().raw('truncate orders, order_items, customers, users, products cascade;'));
  afterAll(() => Order.knex().destroy());

  it("patch an order item", async () => {
    const orderItem = await DataFactory.createOrderItem();

    const response = await request.patch(`/order_items/${orderItem.uuid}`)
      .send({ quantity: 10 })
      .set("Accept", "application/json");

    expect(response.status).toBe(200);
    expect(response.body).toMatchObject({
      id: expect.any(Number),
      created_at: expect.any(String),
      updated_at: expect.any(String),
      uuid: orderItem.order.uuid,
      customer_id: orderItem.order.customer.uuid,
      order_items: expect.arrayContaining([
        expect.objectContaining({
          uuid: expect.any(String),
          created_at: expect.any(String),
          updated_at: expect.any(String),
          product: {
            title: 'Product',
            uuid: orderItem.product.uuid,
            description: "Product description",
            qr: null
          },
          quantity: 10,
          state: "draft"
        })
      ]),
    });
    expect(response.body.order_items.length).toBe(1);
  });

  it("throws 404 for not existing order_item_id", async () => {
    const response = await request.delete(`/order_items/${uuid.v4()}`)
      .send({})
      .set("Accept", "application/json");

    expect(response.status).toBe(404);
  });

  it("throws validation error with additional properties", async () => {
    const orderItem = await DataFactory.createOrderItem();

    const response = await request.patch(`/order_items/${orderItem.uuid}`)
      .send({ quantity: 10, order_id: orderItem.order.id })
      .set("Accept", "application/json");

    expect(response.status).toBe(400);
    expect(response.body).toMatchSnapshot();
  });
});