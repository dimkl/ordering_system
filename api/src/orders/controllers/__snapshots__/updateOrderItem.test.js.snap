// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PATCH /order_items/:order_item_id throws validation error with additional properties 1`] = `
Array [
  Object {
    "instancePath": "",
    "keyword": "additionalProperties",
    "message": "must NOT have additional properties",
    "params": Object {
      "additionalProperty": "order_id",
    },
    "schemaPath": "#/additionalProperties",
  },
]
`;
